strings/base_windows.h

```diff
@@ -128,12 +128,25 @@ namespace winrt::impl
         }
 
         void* result{};
-        hresult code = ptr->QueryInterface(guid_of<To>(), &result);
-        if (code < 0)
+        ptr->QueryInterface(guid_of<To>(), &result);
+        return wrap_as_result<To>(result);
+    }
+
+    template <typename To, typename From, std::enable_if_t<is_com_interface_v<To>, int> = 0>
+    std::pair<com_ref<To>, hresult> try_as_with_reason(From* ptr) noexcept
```

Comment:
Perhaps change `try_as` to be `return try_as_with_reason<T>().first` to reduce code duplication?

Reply:
`as` could also be done this way

Reply:
At this point I've had to tweak `try_as` enough times in a row that I'd be happy to have it back to the exact same code that it had 2 weeks ago.  It's not a long method so the duplication is not too serious, I think.